lib/gps.js

```diff
@@ -0,0 +1,418 @@
+  chip = opts.chip;
+
+  // If a breakout is defined check for receiver and chip
+  if (Breakouts[breakout]) {
+    if (!receiver && Breakouts[breakout].receiver) {
+      receiver = Breakouts[breakout].receiver.value;
+    }
+
+    if (!chip && Breakouts[breakout].chip) {
+      chip = Breakouts[breakout].chip.value;
+    }
+  }
+
+  // If a receiver was defined or derived but chip was not
+  if (receiver && !chip && Receivers[receiver].chip) {
+    chip = Receivers[receiver].chip.value;
+  }
+
+  // Default case
+  if (!chip) {
+    chip = "DEFAULT";
+  }
+
+  // Allow users to pass in custom chip types
+  chip = typeof chip === "string" ?
+    Chips[chip] : opts.chip;
+
+  // Allow users to pass in custom receiver types
+  receiver = typeof receiver === "string" ?
+    Receivers[receiver] : opts.receiver;
+
+  // Chip decorates the instance
+  Object.defineProperties(this, chip);
+
+  // Receiver decorates this instance
+  if (receiver) {
+    Object.defineProperties(this, receiver);
+  }
+
+  // breakout decorates the instance
+  if (opts.breakout) {
+    breakout = typeof opts.breakout === "string" ?
+      Breakouts[opts.breakout] : opts.breakout;
+
+    Board.Controller.call(this, breakout, opts);
+  }
+
+  // If necessary set default property values
+  this.precision = opts.precision || 9;
+  this.baud = opts.baud || breakout.baud || chip.baud;
+
+  // Create a "state" entry for privately
+  // storing the state of the instance
+  state = {
+    sat: {},
+    pos: {},
+    nav: {},
+    frequency: 1000,
+    lowPowerMode: false
+  };
+
+  priv.set(this, state);
+
+  // Getters for private state values
+  Object.defineProperties(this, {
+    pos: {
+      get: function() {
+        return state.pos;
+      }
+    },
+    sat: {
+      get: function() {
+        return state.sat;
+      }
+    },
+    nav: {
+      get: function() {
+        return state.nav;
+      }
+    },
+    time: {
+      get: function() {
+        return state.time;
+      }
+    }
+  });
+
+  if (this.initialize) {
+    this.initialize(opts);
+  }
+
+}
+
+util.inherits(GPS, events.EventEmitter);
+
+/*
+ * Default intialization for serial GPS
+ */
+GPS.prototype.initialize = function(opts) {
+
+  var state = priv.get(this);
+  state.portId = opts.serialPort || opts.portId || opts.port || opts.bus || this.io.SERIAL_PORT_IDs.SW_SERIAL0 || this.io.SERIAL_PORT_IDs[0];
+
+  // Set the pin modes
+  ["tx", "rx"].forEach(function(pin) {
+    if (this.pins[pin]) {
+      this.io.pinMode(this.pins[pin], this.io.MODES.SERIAL);
+    }
+  }, this);
+
+  if (this.pins.onOff) {
+    this.io.pinMode(this.pins.onOff, this.io.MODES.OUTPUT);
+    this.onOff = new Pin(this.pins.onOff);
+  }
+
+  this.io.serialConfig({
+    portId: state.portId,
+    baud: this.baud,
+    rxPin: this.pins.rx,
+    txPin: this.pins.tx
+  });
+
+  if (this.configure) {
+    this.configure(function() {
+      this.listen();
+      if (opts.freq) {
+        this.frequency(opts.freq);
```

Comment:
Just noticed this... I made a mistake 4 years ago in using `freq` as the property name for values that are milliseconds and not hertz. This use is hertz, but has the same name as the mis-named property in every other class. Eventually, I want to fix this throughout, but in the meantime I want to reserve the word "frequency" to be used only as a data property of either the constructor options or a component instance, so it can't be a function (sorry). I will volunteer to make this change if you're not sure what I'm driving at, but let me try to run it down: 

1. Currently, `frequency` is a function, change this to an accessor property. 
2. Currently, `freq` is a property of `opts`, change this to `frequency`
3. Currently, `this.frequency(opts.freq)` is called to send the frequency command and set the initial value of `state.frequency`, change this to `this.frequency = opts.frequency;`